provider/aws/sia-ec2/authn.go

```diff
@@ -85,7 +85,7 @@ func GetECSOnEC2TaskId() string {
 	return taskId
 }
 
-func GetEC2Config(configFile, metaEndpoint string, useRegionalSTS bool, region, account string) (*options.Config, *options.ConfigAccount, error) {
+func GetEC2Config(profileConfigFile, configFile, metaEndpoint string, useRegionalSTS bool, region, account string) (*options.Config, *options.ConfigAccount, *options.AccessProfileConfig, error) {
```

Comment:
we should have profileConfigFile as the second argument - being first implies the most important but it's not - it's an optional setting